File: core/src/main/java/lt/markmerkk/app/box2d/temp_components/wheel/RevolvingWheelImpl.java
package lt.markmerkk.app.box2d.temp_components.wheel;

import com.badlogic.gdx.physics.box2d.World;
import com.badlogic.gdx.physics.box2d.joints.RevoluteJointDef;
import lt.markmerkk.app.box2d.Car;
import org.jetbrains.annotations.NotNull;

public final class RevolvingWheelImpl extends BaseWheelImpl {
   public void initJoint(
           @NotNull World world,
           @NotNull Car car
   ) {
      RevoluteJointDef jointdef = new RevoluteJointDef();
      jointdef.initialize(
              car.getBody(),
              this.getBody(),
              this.getBody().getWorldCenter()
      );
      jointdef.enableMotor = false;
      world.createJoint(jointdef);
   }

   public RevolvingWheelImpl(
           @NotNull World world,
           @NotNull Car car,
           float posX,
           float posY,
           float width,
           float height
   ) {
      super(world, car, posX, posY, width, height, true);
   }
}

File: core/src/main/java/lt/markmerkk/app/box2d/temp_components/wheel/RevolvingWheelImpl.kt
package lt.markmerkk.app.box2d.temp_components.wheel

import com.badlogic.gdx.physics.box2d.*
import com.badlogic.gdx.physics.box2d.joints.RevoluteJointDef
import lt.markmerkk.app.box2d.CarBox2D

/**
 * @author mariusmerkevicius
 * @since 2016-06-05
 */
class RevolvingWheelImpl(
        world: World,
        carBox2D: CarBox2D,
        posX: Float,
        posY: Float,
        width: Float,
        height: Float
) : BaseWheelImpl(world, carBox2D, posX, posY, width, height, true) {

    override fun initJoint(world: World, carBox2D: CarBox2D) {
        val jointdef = RevoluteJointDef()
        jointdef.initialize(carBox2D.body, body, body.worldCenter)
        jointdef.enableMotor = false
        world.createJoint(jointdef)
    }

}